lib/src/services/common-service.dart

```diff
@@ -0,0 +1,131 @@
+part of 'tmdb-service.dart';
+
+abstract class _CommonService<T> {
+  Configuration _configuration;
+  String _apiKey;
+
+  String get type {
```

Comment:
Same thing as above, switch should be changed either to abstract getter with respective implementation in derived classes. Or as an alternative send the url as the parameter (since other method already work in that way, it will be more consistent with the rest of the class)